functions/PushNotificationTutorConfirmation.js

```diff
@@ -0,0 +1,111 @@
+const {onDocumentUpdated} = require("firebase-functions/v2/firestore");
+const {setGlobalOptions} = require("firebase-functions/v2");
+const {initializeApp} = require("firebase-admin/app");
+const {getFirestore} = require("firebase-admin/firestore");
+const admin = require("firebase-admin");
+
+initializeApp();
+
+setGlobalOptions({
+  maxInstances: 10,
+  timeoutSeconds: 60,
+  memory: "256MiB"});
+
+exports.pushNotificationTutorLessonConfirmation = onDocumentUpdated(
+    "lessons/{lessonId}",
+    async (event) => {
+      const before = event.data.before.data(); // Data before the update
+      const after = event.data.after.data(); // Data after the update
+      const lessonId = event.params.lessonId;
+      const db = getFirestore();
+
+      // Check if the status field changed
+      if (before.status !== after.status) {
+        console.log(`Lesson ${lessonId} status changed from ${before.status}
+           to ${after.status}`);
+
+        // Check if the new status is "confirmed"
+        if (before.status === "PENDING_TUTOR_CONFIRMATION" && after.status ===
+           "CONFIRMED") {
+          try {
+            const tutorUid = String(after.tutorUid).split(",")[0];
+            const studentUid = after.studentUid;
+            console.log(`idcollected, ${tutorUid} && ${studentUid}`);
+            const tutorQuery = await db.collection("profiles")
```

Comment:
You can also refactor that section where you handle Firestore interactions, which will make it easier to manage and maintain as we will most likely scale the number of query to Firestore
```javascript
const { getFirestore } = require("firebase-admin/firestore");
const db = getFirestore();

const getUserProfileByUid = async (uid) => {
  const querySnapshot = await db.collection("profiles").where("uid", "==", uid).limit(1).get();
  return querySnapshot.empty ? null : querySnapshot.docs[0];
};

module.exports = {
  getUserProfileByUid,
};